report/renderer/i18n.js

```diff
@@ -22,25 +22,58 @@ export class I18n {
     // When testing, use a locale with more exciting numeric formatting.
     if (locale === 'en-XA') locale = 'de';
 
-    this._numberDateLocale = locale;
-    this._numberFormatter = new Intl.NumberFormat(locale);
-    this._percentFormatter = new Intl.NumberFormat(locale, {style: 'percent'});
+    this._locale = locale;
     this._strings = strings;
   }
 
   get strings() {
     return this._strings;
   }
 
+  /**
+   * @param {number} number
+   * @param {number} granularity
+   * @param {Intl.NumberFormatOptions} opts
+   * @return {string}
+   */
+  _formatNumberWithGranularity(number, granularity, opts = {}) {
+    opts = {...opts};
+    const log10 = -Math.log10(granularity);
+    if (!Number.isFinite(log10) || (granularity > 1 && Math.floor(log10) !== log10)) {
```

Comment:
I think `if (!Number.isFinite(log10) || !Number.isInteger(log10)) {` is sufficient if we just want powers of 10

Reply:
We have a test that uses 0.5 as a value here: https://github.com/GoogleChrome/lighthouse/blob/b4a8e238ccd6593fc3c6f7ba7f60174dd6533e84/report/test/renderer/i18n-test.js#L63

and given we pass `granularity` through a details object throwing here may even be a breaking change... but good thing we are in pre-10.0 mode :)

hmmm throwing in report code is probably bad anyhow. wish we had a DCHECK (cc @exterkamp lol)

Reply:
> We have a test that uses 0.5 as a value here:

ha, well that looks like something that was supported so it was nice to add a test, rather that something we've ever used or wanted to use, but if we want to keep it, `if (!Number.isFinite(log10) || (granularity > 1 && !Number.isInteger(log10)))` is slightly easier to understand

Reply:
for some reason i had no idea `isInteger` existed